internal/prober/prober.go

```diff
@@ -87,36 +94,54 @@ func (p *Prober) Run() {
-	internalShootClient, err := p.setupProbeClient(ctx, p.namespace, p.config.InternalKubeConfigSecretName)
+	p.backOffIfNeeded()
+	shootClient, err := p.setupProbeClient(ctx, p.namespace, p.config.KubeConfigSecretName)
 	if err != nil {
-		p.l.Error(err, "Failed to create shoot client using internal secret, ignoring error, internal probe will be re-attempted")
+		p.l.Error(err, "Failed to create shoot client using the KubeConfig secret, ignoring error, probe will be re-attempted")
 		return
 	}
-	p.probeInternal(internalShootClient)
-	if p.internalProbeStatus.isHealthy(*p.config.SuccessThreshold) {
-		externalShootClient, err := p.setupProbeClient(ctx, p.namespace, p.config.ExternalKubeConfigSecretName)
+	err = p.probeAPIServer(shootClient)
+	if err != nil {
+		p.l.Info("API server probe failed, Skipping lease probe and scaling operation", "err", err.Error())
+		return
+	}
+	p.l.Info("API server probe is successful, will conduct node lease probe")
+
+	candidateNodeLeases, err := p.probeNodeLeases(shootClient)
+	if err != nil {
+		return
+	}
+	if len(candidateNodeLeases) == 0 {
+		p.l.Info("No owned node leases are present in the cluster, skipping scaling operation")
+		return
+	}
+	if p.shouldPerformScaleUp(candidateNodeLeases) {
+		p.l.Info("Lease probe succeeded, performing scale up operation if required")
+		err := p.scaler.ScaleUp(ctx)
 		if err != nil {
-			p.l.Error(err, "Failed to create shoot client using external secret, ignoring error, probe will be re-attempted")
-			return
+			p.l.Error(err, "Failed to scale up resources")
 		}
-		p.probeExternal(externalShootClient)
-		// based on the external probe result it will either scale up or scale down
-		if p.externalProbeStatus.isUnhealthy(*p.config.FailureThreshold) {
-			p.l.Info("External probe is un-healthy, checking if scale down is already done or is still pending")
-			err := p.scaler.ScaleDown(ctx)
-			if err != nil {
-				p.l.Error(err, "Failed to scale down resources")
-			}
-			return
+	} else {
+		p.l.Info("Lease probe failed, performing scale down operation if required")
+		err := p.scaler.ScaleDown(ctx)
+		if err != nil {
+			p.l.Error(err, "Failed to scale down resources")
 		}
-		if p.externalProbeStatus.isHealthy(*p.config.SuccessThreshold) {
-			p.l.Info("External probe is healthy, checking if scale up is already done or is still pending")
-			err := p.scaler.ScaleUp(ctx)
-			if err != nil {
-				p.l.Error(err, "Failed to scale up resources")
-			}
+		return
+	}
+}
+
+// shouldPerformScaleUp returns true if the ratio of expired node leases to valid node leases is less than or equal to
+// the NodeLeaseFailureFraction set in the prober config
+func (p *Prober) shouldPerformScaleUp(candidateNodeLeases []coordinationv1.Lease) bool {
+	var validNodeLeaseCount, expiredNodeLeaseCount float64
+	for _, lease := range candidateNodeLeases {
+		if p.isLeaseExpired(lease) {
+			expiredNodeLeaseCount++
 		}
 	}
+	validNodeLeaseCount = float64(len(candidateNodeLeases))
```

Comment:
a more apt name would be `candidateNodeLeaseCount` or get rid of this altogether. Instead just use 
```go
return expiredNodeLeaseCount/ float64(len(candidateNodeLeases)) < *p.config.NodeLeaseFailureFraction
```